backend/homework/7/src/main/java/org/example/Main.java

```diff
@@ -0,0 +1,64 @@
+package org.example;
+
+
+import logger.Logging;
+import org.example.vehicleservice.config.AppConfig;
+import org.example.vehicleservice.models.Vehicle;
+import org.example.vehicleservice.services.VehicleService;
+import org.example.vehicleservice.utils.MostExpAndCheap;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.stereotype.Component;
+import java.util.List;
+
+@Component
+public class Main {
+
+    private final VehicleService vehicleService1;
+    private final VehicleService vehicleService2;
+
+    @Autowired
+    public Main(@Qualifier("factory1") VehicleService vehicleService1,
+                @Qualifier("factory2") VehicleService vehicleService2) {
+        this.vehicleService1 = vehicleService1;
+        this.vehicleService2 = vehicleService2;
+
+    }
+
+    public void run() {
```

Comment:
Modularize the code into more methods.
Abstract out the null checks and validations